src/indexer.c

```diff
@@ -173,30 +225,82 @@ RedisModuleDict * QueryIndexPredicate(RedisModuleCtx *ctx, QueryPredicate *predi
         if (createResultDict && prevResults == NULL) {
             RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(currentLeaf, "^", NULL, 0);
             RedisModuleString *currentKey;
-            while((currentKey = RedisModule_DictNext(ctx, iter, NULL)) != NULL) {
-                RedisModule_DictSet(localResult, currentKey, (void *)1);
+            while ((currentKey = RedisModule_DictNext(ctx, iter, NULL)) != NULL) {
+                RedisModule_DictSet(localResult, currentKey, (void *) 1);
             }
             RedisModule_DictIteratorStop(iter);
-        }
-        else {
+        } else {
             localResult = currentLeaf;
         }
-    }
+        char *iter_ptr;
+        int nokey;
+
+        const char *key = RedisModule_StringPtrLen(predicate->key, &_s);
+        const char *value;
 
-    if (prevResults != NULL) {
-        if (predicate->type == EQ || predicate->type == CONTAINS) {
-            _intersect(ctx, prevResults, localResult);
-        } else  if (predicate->type == NCONTAINS) {
-            _difference(ctx, prevResults, localResult);
-        } else if (predicate->type == NEQ){
-            _difference(ctx, prevResults, localResult);
+        switch (predicate->type) {
```

Comment:
this is too complex, the easier think to do is:
 RedisModuleDict *localResult = RedisModule_CreateDict(ctx);
for item in predicate-> valuesList:
   _union(localResult, GetDictFromIndex(item))